streams/src/main/java/org/apache/kafka/streams/state/internals/WindowByteProxyStore.java

```diff
@@ -0,0 +1,146 @@
+                    final long windowStartTimestamp) {
+        store.put(key, value);
+    }
+
+    @Override
+    public byte[] fetch(final Bytes key,
+                        final long time) {
+        return convertToTimestampedFormat(store.fetch(key, time));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public WindowStoreIterator<byte[]> fetch(final Bytes key,
+                                             final long timeFrom,
+                                             final long timeTo) {
+        return new WindowIteratorProxy(store.fetch(key, timeFrom, timeTo));
+    }
+
+    @Override
+    public WindowStoreIterator<byte[]> fetch(final Bytes key,
+                                             final Instant from,
+                                             final Instant to) {
+        return new WindowIteratorProxy(store.fetch(key, from, to));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetch(final Bytes from,
+                                                           final Bytes to,
+                                                           final long timeFrom,
+                                                           final long timeTo) {
+        return new KeyValueIteratorProxy<>(store.fetch(from, to, timeFrom, timeTo));
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetch(final Bytes from,
+                                                           final Bytes to,
+                                                           final Instant fromTime,
+                                                           final Instant toTime) {
+        return new KeyValueIteratorProxy<>(store.fetch(from, to, fromTime, toTime));
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> all() {
+        return new KeyValueIteratorProxy<>(store.all());
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetchAll(final long timeFrom,
+                                                              final long timeTo) {
+        return new KeyValueIteratorProxy<>(store.fetchAll(timeFrom, timeTo));
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetchAll(final Instant from,
+                                                              final Instant to) {
+        return new KeyValueIteratorProxy<>(store.fetchAll(from, to));
+    }
+
+    @Override
+    public String name() {
+        return store.name();
+    }
+
+    @Override
+    public void init(final ProcessorContext context,
+                     final StateStore root) {
+        store.init(context, root);
+    }
+
+    @Override
+    public void flush() {
+        store.flush();
+    }
+
+    @Override
+    public void close() {
+        store.close();
+    }
+
+    @Override
+    public boolean persistent() {
+        return true;
```

Comment:
as above

Reply:
As above. Will add a check to the constructor.